File: src/com/github/gtache/contributors/LSPGotoSymbolContributor.scala
package com.github.gtache.contributors

import com.github.gtache.PluginMain
import com.intellij.navigation.NavigationItem
import com.intellij.openapi.project.Project

/**
  * The Go-to contributor for LSP
  */
class LSPGotoSymbolContributor extends LSPGotoContributor {

  override def getItemsByName(name: String, pattern: String, project: Project, includeNonProjectItems: Boolean): Array[NavigationItem] = {
    val res = PluginMain.workspaceSymbols(if (name.endsWith("$")) name.dropRight(1) else name, pattern, project, includeNonProjectItems)
    res
  }


}

File: intellij-lsp/src/main/kotlin/com/github/gtache/lsp/contributors/gotoo/LSPGotoSymbolContributor.kt
package com.github.gtache.lsp.contributors.gotoo

import com.github.gtache.lsp.LSPProjectService
import com.intellij.navigation.NavigationItem
import com.intellij.openapi.components.service
import com.intellij.openapi.project.Project

/**
 * The GotoSymbol contributor for LSP
 */
class LSPGotoSymbolContributor : LSPGotoContributor {

    override fun getItemsByName(name: String, pattern: String, project: Project, includeNonProjectItems: Boolean): Array<NavigationItem> {
        return project.service<LSPProjectService>().workspaceSymbols(name, pattern, includeNonProjectItems)
    }
}